cmd/ledgerutil/main.go

```diff
@@ -41,30 +41,42 @@ func main() {
 		return
 	}
 
-	// Determine result json file location
-	var resultFilepath string
-	if outputDir != nil {
-		resultFilepath = *outputDir
-	} else {
-		resultFilepath, err = os.Getwd()
-		if err != nil {
-			fmt.Println(err)
-			return
-		}
-	}
-	resultFilepath = filepath.Join(resultFilepath, resultFilename)
-
 	// Command logic
 	switch command {
 
 	case compare.FullCommand():
 
-		count, err := ledgerutil.Compare(*snapshotPath1, *snapshotPath2, resultFilepath)
+		cleanpath := filepath.Clean(*outputDir)
+
+		// Determine result json file location
+		if *outputDir == "" {
+			*outputDir, err = os.Getwd()
+			if err != nil {
+				fmt.Printf("%s%s\n", compareErrorMessage, err)
+				return
+			}
+			cleanpath = "the current directory"
+		}
+		// Clean output
+		switch cleanpath {
+		case ".":
+			cleanpath = "the current directory"
+		case "..":
+			cleanpath = "the parent directory"
+		}
+
+		count, err := ledgerutil.Compare(*snapshotPath1, *snapshotPath2, *outputDir)
 		if err != nil {
-			fmt.Println(err)
+			fmt.Printf("%s%s\n", compareErrorMessage, err)
 			return
 		}
-		fmt.Printf("\nSuccessfully compared snapshots. Result saved to %s. Total differences found: %d\n", resultFilepath, count)
+
+		fmt.Print("\nSuccessfully compared snapshots. ")
+		if count == -1 {
+			fmt.Println("Both snapshot public state hashes were the same. No results were generated.")
+		} else {
+			fmt.Printf("Results saved to %s. Total differences found: %d\n", cleanpath, count)
```

Comment:
Now that we set the subdirectory to <channelname>_<height>_comparison, that should be reported here instead of just the cleanpath. That is, return outputDirPath from Compare() so that we can report it here (ideally as an absolute path so that there is no doubt where the output is). That would actually eliminate the need for the cleanpath logic altogether... just report the actual path written including subdirectory.

The rest of the logic looks good, so we can defer that to next PR.